main.py

```diff
@@ -78,8 +79,11 @@ def get_evcs_by_svc_level(self) -> list:
 
         In the future, as more ops are offloaded it should be get from the DB.
         """
-        return sorted(self.circuits.values(),
-                      key=lambda x: (-x.service_level, x.creation_time))
+        with self._lock_circuits:
```

Comment:
@Alopalao, a list comprehension here filtering should suffice making it easier to follow, something like:

```
    return sorted(
        [circuit for circuit in self.circuits.values() if circuit.is_enabled()],
        key=lambda x: (-x.service_level, x.creation_time),
    )
```

What you wrote is also right, but no need for a dict comprehension here.

Also, you can also nuke `self._lock_circuits` no need here, `sorted()` is already making a shallow copy so at least in terms of iteration we won't have thread issues. But then if callers need to perform a side even then they'll acquire `evc.lock` (which you've already covered). See what I mean?

Reply:
Replace `lock` dynamic. Added argument to decided whether to filter or not because [consistency_check](https://github.com/kytos-ng/mef_eline/blob/820d024b3a1c861fbd4ad9fbbe2cb31301b31078/main.py#L138-L140) complains if it does not get all evcs.

Reply:
Indeed. Good catch, yes, consistency check is also checking if whatever is expected to be in memory is indeed in memory including disabled EVCs.